scipy/sparse/csgraph/tests/test_graph_laplacian.py

```diff
@@ -34,26 +45,25 @@ def _check_symmetric_graph_laplacian(mat, normed):
     else:
         sp_mat = sparse.csr_matrix(mat)
 
-    laplacian = csgraph.laplacian(mat, normed=normed)
     n_nodes = mat.shape[0]
-    if not normed:
-        assert_array_almost_equal(laplacian.sum(axis=0), np.zeros(n_nodes))
-    assert_array_almost_equal(laplacian.T, laplacian)
-    assert_array_almost_equal(
-        laplacian, csgraph.laplacian(sp_mat, normed=normed).toarray())
-
-    assert_array_almost_equal(laplacian,
-            _explicit_laplacian(mat, normed=normed))
-
-
-def test_laplacian_value_error():
-    for t in int, float, complex:
-        for m in ([1, 1],
-                  [[[1]]],
-                  [[1, 2, 3], [4, 5, 6]],
-                  [[1, 2], [3, 4], [5, 5]]):
-            A = np.array(m, dtype=t)
-            assert_raises(ValueError, csgraph.laplacian, A)
+    explicit_laplacian = _explicit_laplacian(mat, normed=normed)
+    laplacian = csgraph.laplacian(mat, normed=normed, copy=copy)
+    sp_laplacian = csgraph.laplacian(sp_mat, normed=normed,
+                                     copy=copy)
+    assert_array_almost_equal(laplacian, sp_laplacian.toarray())
+
+    for tested in [laplacian, sp_laplacian.toarray()]:
+        if not normed:
+            assert_array_almost_equal(tested.sum(axis=0), np.zeros(n_nodes))
+        assert_array_almost_equal(tested.T, tested)
+        assert_array_almost_equal(tested, explicit_laplacian)
+
+    if not copy:
```

Comment:
If I am not mistaking, there is no test checking that the original `mat` is unaltered when `copy=True`. You can do a copy as the first step to store the original value to compare with. As for the False case, the evidence that the array is different should happen before any other condition.

Reply:
Correct. I have added a check for `copy=True` and swapped the order

Reply:
@tupui Done. The failures appear unrelated to this PR.